### ISSUE TITLE
Query Params in URL, or ?

### ISSUE THREAD

**Issue by NadavK** [shellinabox/shellinabox#376]
In my solution, shellinabox is embedded within a web-portal that already contains the user's ssh credentials, so I don't want the user to have to re-enter them in shellinabox (similar to SSO single-sign-on).
Is it possible for shellinabox to login with username+password that it receives via url? Or some other way?

(sorry for opening an Issue - I could find no other way of posting a question)

**Comment by guanglinlv** [comment]
you have a workaround,

```
./shellinaboxd -t -d -n -s /shell:root:root:HOME:'sshpass -p PASSPHRASE ssh -t USER@HOSTNAME'
```

or you can modify the code on the definition of service. i had did it, if you want to use it, i can fork and push it.

**Comment by guanglinlv** [comment]
of course, if you want to pass the user and passphrase in you url queries but not the definition of service.  you can see this commit in the other fork.
[Add auto-login feature](https://github.com/anilgulecha/shellinabox/commit/66335a31bb8ac00c3aae521cbaa1a69555551072), it modified the `root_page.html` simply. but, i think it's very dangerous.

**Comment by NadavK** [comment]
[Auto-login](https://github.com/anilgulecha/shellinabox/commit/66335a31bb8ac00c3aae521cbaa1a69555551072) is exactly what I was looking for.
Thanks so much!

**Comment by uudruid74** [comment]
FYI - shellinabox can pass the URL in ${url} and be configured to run anything you want.   I had an interesting hack where specific buttons on the web page would open a text-based application in a running tmux in the shell window.  If the application was already running, it switches to it.  This was done by having shellinabox run a shell script that parsed the URL.   It was for a project where security wasn't an issue, but you can get some ideas from it https://github.com/uudruid74/shellinabox_addons

**Comment by FalcoSuessgott** [comment]
Hello.
Im Sorry, but I do not understand the '-s' parameter. I want to pass arguments in the URL, so that it will start a script on the Server where Shellinabox is hosted.
Could someone provide me a simple example on how to start the shellinabox-daemon and how the URL have to be, so that this should work?

**Comment by uudruid74** [comment]
Check my github for shellinabox_addons which uses this parameter and shows how to parse it server side.   Thats a complete example with everything working of exactly this parameter.

On Aug 29, 2017, FalcoSuessgott <notifications@github.com> wrote:
>Hello.
>Im Sorry, but I do not understand the '-s' parameter. I want to pass
>arguments in the URL, so that it will start a script on the Server
>where Shellinabox is hosted.
>Could someone provide me a simple example on how to start the
>shellinabox-daemon and how the URL have to be, so that this should
>work? 

-- Sent with K-@ Mail - the evolution of emailing.

**Comment by FalcoSuessgott** [comment]
Thanks for your answer.
All I need is to echo the url-parameter in a script.
So currently I call the SIAB daemon like this:

`sudo shellinaboxd  -s /urlargs/:$USER:$GROUP:$HOME:/path/to/script/script.sh ${url}`

my expected output of the called script would be 

`urlargs`

Unfortunately this doesn't work, do you have an advice ?

thanks in advance

**Comment by uudruid74** [comment]
I'm not sure if arguments are supported as I've not looked at it in awhile.  i know that passing the URL in the environment works.  See the shellinabox script in my addon github.   It's an init.d script, but you can see how it builds the command parameter and you can see the script it runs.  It's all there.

On Aug 30, 2017, FalcoSuessgott <notifications@github.com> wrote:
>Thanks for your answer.
>All I need is to echo the url-parameter in a script.
>So currently I call the SIAB daemon like this:
>
>`sudo shellinaboxd  -s
>/urlargs/:$USER:$GROUP:$HOME:/path/to/script/script.sh ${url}`
>
>my expected output of the called script would be 
>
>`urlargs`
>
>Unfortunately this doesn't work, do you have an advice ?
>
>thanks in advance

-- Sent with K-@ Mail - the evolution of emailing.

**Comment by paulbsch** [comment]
Here is how I'm accepting the password from the URL.  This can be easily modified to also allow the user name in the URL.

Here's the command to write to the shellinaboxd config (on CentOS):
```
cat <<EOF | sudo tee -a /etc/sysconfig/shellinaboxd >/dev/null
OPTS="--disable-ssl-menu -s '/:root:root:/:/usr/local/shellinabox/cmd.sh \"\\\${url}\"'"
EOF
```
Here's the code for cmd.sh:
```
#!/bin/bash

URL="$1"
PASS="$(python -c "import urlparse;p=urlparse.urlparse(\"$URL\");print urlparse.parse_qs(p.query)[\"password\"][0];")"
USER="<username>"

sshpass -p "$PASS" ssh -o StrictHostKeyChecking=no -t "$USER@localhost"
```
After starting/restarting shellinaboxd, you'll be able to log in via:
`https://hostname:4200/?password=<user-password>`

I am using this from a private application.  But, in general, this is obviously dangerous if you're not careful.

**Comment by qbeukes** [comment]
wow thanks for sharing!

**Comment by Phrogfur** [comment]
In reference to Aguanglinlv's add auto-login feature, which modifies the root_page.html 
I figured out how to recompile shellinabox 

However I thought I could pass http:\\192.168.1.99:4200/?autologinuser=spud&autologinpassword=yada

Or am I missing something?

I understand how dangerous this.

# Here is the instructions for recompiling shellinabox that seemed  
# to work for me on Centos 6.9
# To rebuild src.rpm --> rpm in one command
# rpmbuild --rebuild shellinabox-2.20-1.el5.src.rpm
#-------------------------------------------------------
# Modifying instructions
# Extract src
rpm -ivh shellinabox-2.20-1.el5.src.rpm
cd rpmbuild/SOURCES
tar -zxvf shellinabox-2.20.tar.gz
rm shellinabox-2.20.tar.gz
>>>MODIFY CODE<<<
tar -czvf shellinabox-2.20.tar.gz shellinabox-2.20
cd ..
rpmbuild -ba SPECS/shellinabox.spec
SEE RPMS dir

**Comment by Phrogfur** [comment]
This is root_page.html that works 

[https://raw.githubusercontent.com/anilgulecha/shellinabox/6837748b4fc7cd28d721518ec22e6d06ff0e8b66/shellinabox/root_page.html](url)
